### PR TITLE
Add rollbacks and info messages

### PR DESCRIPTION
**Pull request by ruiasribeiro**

### CODE REVIEW

**Comment by ruiasribeiro** [comment]
This isn't actually as linear as I thought it would be. The `assert` statements in the original transactions can also cause them to stop early, so we'll also need to rollback in those cases. 

Shouldn't be too hard. I've got an idea on how to solve it, but for now I'll set this PR as a draft.

**Comment by ruiasribeiro** [comment]
Using asserts for non-critical exits (like cancelling the cart approval because it's empty) on the new transactions was creating additional problems, so I ended up reverting to something akin to what I was doing earlier.

To sum up:
- if there's a critical error, like the DB connection does not exist, an exception should be raised, either manually with `raise` or with an `assert`;
- if not, simply rolling back the transaction with `rollbackTransaction()` and returning from the method will work.

File: pytpcc/drivers/postgresdriver.py [hunk 1]
```diff
@@ -255,10 +255,21 @@ def executeTransaction(self, txn, params):
         try:
             super().executeTransaction(txn, params)
         except (Exception, AssertionError) as ex:
-            if self.conn is not None:
-                logging.debug(f"Rolling back {txn}...")
-                self.conn.rollback()
-            raise ex
+            self.rollbackTransaction()
+            raise
```
Comment (Adri029) [inline pytpcc/drivers/postgresdriver.py:259]: Shouldn't this be `raise ex`?
Or is just raise enough?

Comment (ruiasribeiro) [reply pytpcc/drivers/postgresdriver.py:259]: Yeah, it is enough. `raise` by itself re-raises the exception :smile:

**Pull request merged**